src/NuGet.Core/NuGet.Build.Tasks.Console/NuGet.Build.Tasks.Console.csproj

```diff
@@ -34,6 +34,7 @@
     <PackageReference Include="Microsoft.Build.Utilities.Core" ExcludeAssets="Runtime" />
     <!-- System.Security.Cryptography.Xml is a dependency of Microsoft.Build.Tasks.Core. Therefore, we have assigned the same ExcludeAssets value. -->
     <PackageReference Include="System.Security.Cryptography.Xml" ExcludeAssets="Runtime" />
+    <PackageReference Include="System.Security.Cryptography.Pkcs" ExcludeAssets="runtime" />
```

Comment:
Since NuGet.Build.Tasks.Console has a PackageReference on NuGet.Buld.Tasks, it shouldn't be necessary to add this package reference in both. Adding to the other one will cause this project, .Console, to get the higher version transitively.

Reply:
I tried to remove this one then GC alert popped up again, see [here](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8061706&view=logs&j=cb98efd6-7062-56ef-f642-0176540bbe0a&t=f95ba3d8-e30a-5687-3fc7-3bc14bb0710b&l=2339).

Reply:
Ah, the unused package CG alerts. 
I reached out to the CG folks again. We had a conversation about that some time ago, I'm gonna try to revive it.

Reply:
Could we modify the NuGet dependency graph calculation to prevent the download of unused packages like this? While it might be tricky, I believe it's still doable. It might not be ideal as a default experience, but it could be offered as an opt-in feature.

Reply:
It's actually a super easy change, but it makes all subsequent true restore slower as it'd need to redownload packages to compute the graph.

Reply:
Is `runtime` supposed to be there?

Can we get a `before` and  `after` of the deps.json file?

That's the most fullproof way of ensuring we didn't change something major.

Reply:
Is it ok I get it from my local? Or does it have to be from CI pipeline?

Reply:
Here is comparison from my local build with debug config, I can see 3 changes:
Left side is from this branch, right side is from dev branch.
![image](https://github.com/NuGet/NuGet.Client/assets/8766776/65588145-9851-4ede-bd10-11b5a74ce861)
![image](https://github.com/NuGet/NuGet.Client/assets/8766776/d9b95977-6adb-468a-bd63-4adb01a59669)
![image](https://github.com/NuGet/NuGet.Client/assets/8766776/b8570b69-8723-4785-84cd-b1eede3d7bd1)

Reply:
Local is fine. 

Seems like we're missing the PKCS.dll on the left side, which could be an issue if pkcs.dll for whatever reason doesn't get composed into the SDK. 

tldr; We probably have an extra `exclude=runtime` somewhere in the graph.

Reply:
this `PackageReference` is still using `ExcludeAssets="runtime"`. It's only the other file that was changed to `compile`.